guardian/support-service-lambdas
# Remove CorporateAccountId__c from Zuora DataLake Export query

## What does this change?
Removes the `CorporateAccountId__c` field from the `Subscription` export query. This was causing an error from Zuora:

```
"errorCode": "90005",
    "message": "There is a syntax error in one of the queries in the AQuA input (Error = There is no field named CorporateAccountId__c, Query = SELECT AutoRenew, CancellationReason__c, ContractAcceptanceDate, ContractEffectiveDate, IPCountry__c, CreatedDate, Name, InitialPromotionCode__c, PromotionCode__c, ReaderType__c, Status, TermEndDate, TermStartDate, Version, serviceActivationDate, ID, BillToContact.ID, SoldToContact.ID, SubscriptionVersionAmendment.ID, Account.ID, AcquisitionCase__c, AcquisitionSource__c, ActivationDate__c, CanadaHandDelivery__c, CancelledDate, CASSubscriberID__c, CreatedByCSR__c, CreatedByID, CreatedRequestId__c, CreatorAccountID, CreatorInvoiceOwnerID, CurrentTerm, CurrentTermPeriodType, Gift_Subscription__c, InitialTerm, InitialTermPeriodType, InvoiceOwnerID, IPAddress__c, IsInvoiceSeparate, LastPriceChangeDate__c, legacy_cat__c, LegacyContractStartDate__c, CorporateAccountId__c, RedemptionCode__c, GifteeIdentityId__c, GiftRedemptionDate__c, GiftNotificat",
```

## How to test
Run the zuora-datalake-export in DEV and verify that the Lambda can complete all queries without this error (the field has also been deleted in DEV).

## How can we measure success?
The Lambda will run without errors.

## Have we considered potential risks?
This has been alarming since 7th April, so if there is a follow-on issue any impact is likely to be low (even noticeable at all). We believe this might be part of a deprecated data pipeline, and will be clarifying with relevant teams.
